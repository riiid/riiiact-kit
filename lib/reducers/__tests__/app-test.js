import reducer from 'reducers/app';
import {
  toggleDrawer,
  showSnackbar, hideSnackbar,
  showProgress, hideProgress,
  showToolbar, hideToolbar
} from 'actions/app';

describe('reducers/app', () => {

  it('should return initial state', () => {
    expect(reducer(undefined, {})).toEqual({
      drawerOpen: false,
      snackbar: {
        message: '',
        open: false
      },
      toolbar: true
    });
  });

  it('should handle toggleDrawer()', () => {
    expect(reducer(undefined, toggleDrawer())).toEqual({
      drawerOpen: true,
      snackbar: {
        message: '',
        open: false
      },
      toolbar: true
    });
  });

  it('should handle showSnackbar()', () => {
    expect(reducer(undefined, showSnackbar('MESSAGE', {prop: 'PROP'}))).toEqual({
      drawerOpen: false,
      snackbar: {
        message: 'MESSAGE',
        prop: 'PROP',
        open: true
      },
      toolbar: true
    });
  });

  it('should handle hideSnakcbar()', () => {
    expect(reducer(undefined, hideSnackbar())).toEqual({
      drawerOpen: false,
      snackbar: {
        message: '',
        open: false
      },
      toolbar: true
    });
  });

  it('should handle showProgress()', () => {
    expect(reducer(undefined, showProgress())).toEqual({
      drawerOpen: false,
      progress: true,
      snackbar: {
        message: '',
        open: false
      },
      toolbar: true
    });
  });

  it('should handle hideProgress()', () => {
    expect(reducer(undefined, hideProgress())).toEqual({
      drawerOpen: false,
      progress: false,
      snackbar: {
        message: '',
        open: false
      },
      toolbar: true
    });
  });

  it('should handle showToolbar()', () => {
    expect(reducer(undefined, showToolbar())).toEqual({
      drawerOpen: false,
      snackbar: {
        message: '',
        open: false
      },
      toolbar: true
    });
  });

  it('should handle hideToolbar()', () => {
    expect(reducer(undefined, hideToolbar())).toEqual({
      drawerOpen: false,
      snackbar: {
        message: '',
        open: false
      },
      toolbar: false
    });
  });
});
