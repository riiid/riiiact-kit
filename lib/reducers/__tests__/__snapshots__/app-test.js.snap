// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/app should handle hideProgress() 1`] = `
Object {
  "drawerOpen": false,
  "progress": false,
  "snackbar": Object {
    "message": "",
    "open": false,
  },
  "toolbar": true,
}
`;

exports[`reducers/app should handle hideSnakcbar() 1`] = `
Object {
  "drawerOpen": false,
  "snackbar": Object {
    "message": "",
    "open": false,
  },
  "toolbar": true,
}
`;

exports[`reducers/app should handle hideToolbar() 1`] = `
Object {
  "drawerOpen": false,
  "snackbar": Object {
    "message": "",
    "open": false,
  },
  "toolbar": false,
}
`;

exports[`reducers/app should handle showProgress() 1`] = `
Object {
  "drawerOpen": false,
  "progress": true,
  "snackbar": Object {
    "message": "",
    "open": false,
  },
  "toolbar": true,
}
`;

exports[`reducers/app should handle showSnackbar() 1`] = `
Object {
  "drawerOpen": false,
  "snackbar": Object {
    "message": "MESSAGE",
    "open": true,
    "prop": "PROP",
  },
  "toolbar": true,
}
`;

exports[`reducers/app should handle showToolbar() 1`] = `
Object {
  "drawerOpen": false,
  "snackbar": Object {
    "message": "",
    "open": false,
  },
  "toolbar": true,
}
`;

exports[`reducers/app should handle toggleDrawer() 1`] = `
Object {
  "drawerOpen": true,
  "snackbar": Object {
    "message": "",
    "open": false,
  },
  "toolbar": true,
}
`;

exports[`reducers/app should return initial state 1`] = `
Object {
  "drawerOpen": false,
  "snackbar": Object {
    "message": "",
    "open": false,
  },
  "toolbar": true,
}
`;
